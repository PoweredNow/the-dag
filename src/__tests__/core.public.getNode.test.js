const addEdges = require('../core/public/addEdges');
const addNodes = require('../core/public/addNodes');
const getNode = require('../core/public/getNode');
const _createNode = require('../core/private/_createNode');

describe('core.public.getNode', () => {
  test('exports', () => {
    expect(getNode).toMatchSnapshot();
  });
  test('Gets node', () => {
    addNodes([{ nodeID: 1 }, { nodeID: 2 }]);
    addEdges([{ source: { nodeID: 1 }, target: { nodeID: 2 } }]);
    expect(getNode({ nodeID: 1 })).toMatchSnapshot();
  });
  test('Creates id for node if id not passed in private createNode', () => {
    const node = _createNode({ data: {} });
    expect(node).toMatchSnapshot();
  });
});
