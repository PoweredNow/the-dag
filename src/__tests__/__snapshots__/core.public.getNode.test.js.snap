// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.public.getNode Creates id for node if id not passed in private createNode 1`] = `
Object {
  "nodeData": Object {},
  "nodeID": "FAKE_UUID_V4",
  "possibleTargets": Array [],
}
`;

exports[`core.public.getNode Gets node 1`] = `
Object {
  "nodeData": undefined,
  "nodeID": 1,
  "possibleTargets": Array [
    2,
  ],
}
`;

exports[`core.public.getNode exports 1`] = `[Function]`;
